frontend/src/lib/api.ts

```diff
@@ -1232,8 +1232,12 @@ const api = {
         ): Promise<NotebookType> {
             return await new ApiRequest().notebook(notebookId).update({ data })
         },
-        async list(): Promise<PaginatedResponse<NotebookType>> {
-            return await new ApiRequest().notebooks().get()
+        async list(sessionRecordingId?: string): Promise<PaginatedResponse<NotebookType>> {
+            const apiRequest = new ApiRequest().notebooks()
+            if (sessionRecordingId) {
+                apiRequest.withQueryString({ has_recording: sessionRecordingId })
```

Comment:
Yeah, I quite liked the `has_recording: string|true|false` type for being concise.

But then if it's stood out once maybe it's the wrong call. What would your expectation be?

has_recording: boolean
session_recording_id: string

??

Reply:
Initially yes, but now that you've explained it that makes sense.

Given we'll probably want to be able to search notebooks for more than just recordings it's probably better not to have two separate params or we'll end up with double the amount of params for each node type we allow.

In that vein, is there a different way we could let people search for content? Conscious we might need to update the API every time we add a new node otherwise. Maybe something like `?contains=recording:12345678` similar to GitHub searches. Just a thought